File: packages/core/src/engine/verbs/intersect.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import ColumnTable from 'arquero/dist/types/table/column-table'
import { TableStore } from '../../index.js'
import { SetOp, Step } from '../../types.js'
import { set } from '../util/index.js'

/**
 * Executes an arquero table intersect.
 * @param step
 * @param store
 * @returns
 */
export async function intersect(
	step: Step,
	store: TableStore,
): Promise<ColumnTable> {
	return set(step, store, SetOp.Intersect)
}

File: python/data-wrangling-components/data_wrangling_components/engine/verbs/intersect.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

from typing import List

import pandas as pd

from data_wrangling_components.table_store import TableContainer


def intersect(source: TableContainer, others: List[TableContainer]):
    input_table = source.table
    others = [other.table for other in others]
    others = pd.concat(others)

    output = input_table.merge(others, how="left", indicator=True)
    output = output[output["_merge"] == "both"]
    output = output.drop("_merge", axis=1).reset_index(drop=True)

    return TableContainer(table=output)
